File: C#/Leetcode/LinkedList/ReverseLinkedList.cs
using LeetcodeSolutions.DataStructures;

namespace LeetcodeSolutions.LinkedList
{
    // Leetcode 206  - https://leetcode.com/problems/reverse-linked-list/
    // Submission Detail - https://leetcode.com/submissions/detail/140383800/

    public class ReverseLinkedList
    {
        // Tx = O(n)
        // Sx = O(1)
        // Inplace algorithm
        public ListNode ReverseList(ListNode head)
        {
            ListNode current = head;
            ListNode previous = null;
            ListNode nextNode = null;

            while (current != null)
            {
                nextNode = current.Next;
                current.Next = previous;
                previous = current;
                current = nextNode;
            }

            return previous;
        }
    }
}

File: Java/LinkedList/ReverseLinkedList.java
package LinkedList;

import DataStructures.ListNode;

public class ReverseLinkedList {
  // Leetcode 206: https://leetcode.com/problems/reverse-linked-list/description/
  // Submission Detail: https://leetcode.com/problems/reverse-linked-list/submissions/1631625777

  //  Input: head = [1,2,3,4,5]
  //  Output: [5,4,3,2,1]

  public ListNode reverseList(ListNode head) {
    ListNode prev = null, current = head, next = null;

    while(current != null) {
      next = current.next;
      current.next = prev;
      prev = current;
      current = next;
    }

    return prev;
  }
}
